server/service/appconfig.go

```diff
@@ -376,7 +376,7 @@ func (svc *Service) ModifyAppConfig(ctx context.Context, p []byte, applyOpts fle
 		if appConfig.SMTPSettings.SMTPEnabled {
 			if oldSMTPSettings != *appConfig.SMTPSettings || !appConfig.SMTPSettings.SMTPConfigured {
 				if err = svc.sendTestEmail(ctx, appConfig); err != nil {
-					return nil, ctxerr.Wrap(ctx, err)
+					return nil, fleet.NewInvalidArgumentError("SMTP Options", err.Error())
```

Comment:
Does `InvalidArgumentError` implement `StatusCode`? (I cannot find it.)
Also, let's add a integration test.

Reply:
Ah, it implements `validationErrorInterface` which will make it return `http.StatusUnprocessableEntity`. So I think we are good.

We are just missing the test.